test/unit/communication/test_async.cpp

```diff
@@ -0,0 +1,43 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <bits/c++config.h>
+#include <gtest/gtest.h>
+#include <mpi.h>
+#include <hpx/lcos/wait_all.hpp>
+#include <vector>
+
+#include "dlaf/communication/async.h"
```

Comment:
This header is the one under test and should be put as first, before any other header.